design/one-pager-component-mutable-and-versioning.md

```diff
@@ -0,0 +1,477 @@
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components:
+    - componentName: frontend
+      traits:
+        - trait:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: FancyTrait
+            spec:
+              traffic:
+                - revision: frontend-v0.0.2
+                  percent: 80%
+                - revision: latest
+                  percent: 20%                 
+```
+
+The real workload instance will be created with the same name as the RevisionName.
+
+```yaml
+  apiVersion: core.oam.dev/v1alpha2
+  kind: ContainerizedWorkload
+  metadata:
+    name: frontend-v0.0.1
+  spec:
+    containers:
+    - name: my-cool-workload
+      image: example/very-cool-workload:0.1.2@sha256:verytrustworthyhash
+      cmd:
+      - "bash lscpu"
+```
+
+**NOTICE:** In this case, many workload instances of `frontend` will be created by OAM runtime.
+The `FancyTrait` should ensure only the latest workload and the one whose name is `frontend-v0.0.1` can be running. 
+
+### Always Using the latest revision when using componentName field
+
+When we use a Component with componentName in ApplicationConfiguration, OAM runtime will always use the latest
+component revision to update/create workload and all corresponding traits. 
+
+For example, below is an AppConfig containing one component with two traits.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: TraitDefinition
+metadata:
+  name: rollouts.core.oam.dev
+spec:
+  revisionAware: true
+  definitionRef:
+    name: rollouts.core.oam.dev
+---
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components:
+    - componentName: frontend
+      traits:
+        - trait:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: ManuelScaler
+            spec:
+              replica: 3
+        - trait:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: Rollout
+            spec:
+              source:
+                revision: frontend-v0.0.1 
+```
+
+Once the `frontend` component changed, the `ApplicationConfiguration` controller will trigger
+an upgrade for the application.
+
+1. Because the component has a revisionAware trait binding with it, so we will aways create new workload instead of update. 
+2. Create a workload according to the new component revision. 
+3. Update all traits pointing their target workloadRef to the new workload.
+4. Rollout trait will do rolling update from `frontend-v0.0.1` to the new one, and finally delete the source workload.
+
+
+### Using revisionName field can specify a fixed component version
+
+When a `ComponentRevision` is specified in ApplicationConfiguration like below:
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components:
+    - revisionName: frontend-v0.0.1
```

Comment:
Can we use two attributes instead: name and version?

Reply:
same  with https://github.com/crossplane/oam-kubernetes-runtime/pull/29#discussion_r433637657